### PR TITLE
Update to use the new null-aware subscript syntax.

### PR DESCRIPTION
**Pull request by munificent**

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ environment:
   sdk: '>=2.3.0 <3.0.0'
```
Comment (kevmoo) [inline pubspec.yaml:10]: also need to bump the min SDK support?

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 language: dart
 
 dart:
-  - 2.3.0
```
Comment (kevmoo) [inline .travis.yml:4]: It's good to keep this in sync w/ the lowerbound of the pubspec.yaml sdk constraint

File: pubspec.yaml (hunk 1 above)
Comment (munificent) [reply pubspec.yaml]: OK, I'll try 2.7.0 and see if that satisfies Travis and the version solver. (The reason I bumped this is that there's no analyzer package that works on 2.3.0 and meets dart_style's new higher constraint.)

File: .travis.yml (hunk 2 above)
Comment (munificent) [reply .travis.yml:4]: Good to know. Done.

**Comment by kevmoo** [comment]
It keeps us honest to make sure we test the lower bound of the SDK that we
claim to support!

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@
 * Always place a blank line after script tags (#782).
 * Don't add unneeded splits on if elements near comments (#888).
 * Indent blocks in initializers of multiple-variable declarations.
+* Update the null-aware subscript syntax from `?.[]` to `?[]`.
```
Comment (kevmoo) [inline CHANGELOG.md:12]: tis nice to tell folks you bumped the min SDK here, too! 😊

Comment (munificent) [reply CHANGELOG.md:12]: This doesn't seem CHANGELOG worthy to me.

**Pull request merged**